dbms/src/Interpreters/Join.cpp

```diff
@@ -739,19 +775,14 @@ std::unique_ptr<IColumn::Offsets> NO_INLINE joinRightIndexedColumns(
                 auto & mapped = find_result.getMapped();
 
                 if constexpr (STRICTNESS == ASTTableJoin::Strictness::Asof)
-                {
-                    Join::AsofGetterType asof_getter(asof_column);
-                    auto asof_key = asof_getter.getKey(i, pool);
-                    bool actually_found = addFoundRowAsof<Map>(mapped, added_columns, current_offset, asof_key);
-
-                    if (actually_found)
+                    if (const Join::RowRef * found = mapped.findAsof(asof_column, i, pool))
```

Comment:
It's better to safe braces for if constexpr (). It's not simple one line statement.